### PR TITLE
feat: Improve metrics stress test handling

### PR DESCRIPTION
**Pull request by bradbeam**
While testing this I think I hit some timing issues that caused this to fail unexpectedly.
I also started getting some race issues with the test backend.

Signed-off-by: Brad Beam <brad.beam@b-rad.info>

### CODE REVIEW

File: httpx/metrics_test.go [hunk 1]
```diff
@@ -166,88 +166,91 @@ func TestStressMetrics(t *testing.T) {
...
-							Method:       http.MethodGet,
-							URL:          newURL("/test"),
-						}
-
-						TimeoutSentMutex.Lock()
-						numTimeoutsSent++
-						TimeoutSentMutex.Unlock()
-					}
+			for msgCount := 0; msgCount < numMessagesPerSender; msgCount++ {
+				connID := ksuid.New().String()
```
Comment (andrewhare) [inline httpx/metrics_test.go:201]: One thing we might want to consider doing is have this be configurable to simulate long-running connections as well.

**Review by andrewhare — APPROVED** [review]
Looks good

File: httpx/metrics_test.go [hunk 2]
```diff
@@ -166,88 +166,91 @@ func TestStressMetrics(t *testing.T) {
...
+
+				// Send Response
+				metrics.In() <- &Response{
+					ConnectionID: connID,
+					ExchangeID:   eID,
+					Created:      refTime.Add(10 * time.Millisecond),
+					StatusCode:   http.StatusOK,
+				}
+
+				messageCountCh <- struct{}{}
```
Comment (ztreinhart) [inline httpx/metrics_test.go:227]: Oh, I like this way of handling complete exchanges vs. timeouts. Nice improvement.

File: httpx/metrics_test.go [hunk 3]
```diff
@@ -166,88 +166,91 @@ func TestStressMetrics(t *testing.T) {
...
+		case err := <-errs:
+			assert.NilError(t, err)
+		case <-time.After(testTimeout):
+			break
+		}
+
+		if numExchangesSent+numTimeoutsSent == int64(numSenders*numMessagesPerSender) {
+			break
+		}
+	}
```
Comment (ztreinhart) [inline httpx/metrics_test.go:248]: Clever way to avoid using mutexes! I'll have to remember this one for the future.

**Review by ztreinhart — APPROVED** [review]
Very nice!

**Pull request merged**